### PR TITLE
Make MessageAcknowledger concrete

### PR DESCRIPTION
**Pull request by fuchsnj**

### CODE REVIEW

File: src/main/java/com/bandwidth/sqs/consumer/acknowledger/MessageAcknowledger.java [hunk 1]
```diff
@@ -135,6 +141,10 @@ public Completable getCompletable() {
         return ackingComplete;
     }
 
+    public MessagePublisher<T> getMessagePublisher(){
```
Comment (derelbenkoenig) [inline src/main/java/com/bandwidth/sqs/consumer/acknowledger/MessageAcknowledger.java]: why is a getter needed for this?

File: src/main/java/com/bandwidth/sqs/consumer/acknowledger/MessagePublisher.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package com.bandwidth.sqs.consumer.acknowledger;
+
+import java.time.Duration;
+
+import io.reactivex.Completable;
+
+public interface MessagePublisher<T> {
```
Comment (derelbenkoenig) [inline src/main/java/com/bandwidth/sqs/consumer/acknowledger/MessagePublisher.java]: Does this library not already have a `Publisher` class of some kind?

File: src/main/java/com/bandwidth/sqs/consumer/acknowledger/MessageAcknowledger.java (hunk 1 above)
Comment (fuchsnj) [inline src/main/java/com/bandwidth/sqs/consumer/acknowledger/MessageAcknowledger.java]: It makes it easy to wrap the Acknowledger in a different type. A JSON acknowledger could convert a JSON object to a normal SQS Message, this use this publisher to actually publish. (The could that would get the publisher doesn't have access to the publisher provided in the constructor)

File: src/main/java/com/bandwidth/sqs/consumer/acknowledger/MessagePublisher.java (hunk 2 above)
Comment (fuchsnj) [inline src/main/java/com/bandwidth/sqs/consumer/acknowledger/MessagePublisher.java]: Not yet, it has a "Client" but that doesn't know how to publish an arbitrary `T` type

**Pull request closed without merging**